---
layout: post
title: "TIL - 0107"
date: 2019-01-07 00:00:00
img:
categories:
- TIL
tags: [TIL]
---

## 전문가를 위한 파이썬

---
## 15.1
- for : for 루프 실행 후, else
- while : while 루프 ㅃ빠져나온 후에 else
- try : try 블록에서 예외가 발생하지 않을때 else
- try .. finally :try문에는 finally절을 사용할 수 있다. finally절은 try문 수행 도중 예외 발생 여부에 상관없이 항상 수행된다. 보통 finally절은 사용한 리소스를 close해야 할 경우에 많이 사용된다.
```python
try:
  danagerous_call()
  after_call()
except OSError:
  log(..)

try:
  danagerous_call()
except OSError:
  log(..)
else:
  after_call()
```

- [try-else-except 에 대한 읽을거리](https://stackoverflow.com/questions/16138232/is-it-a-good-practice-to-use-try-except-else-in-python)
